index.js

```diff
@@ -455,35 +456,152 @@ var VectorWatchStream = function () {
         }
     }
 
-    /** Removes the received settings object overheard so the developer only handles the settings themselves.
-     * @param settingsMap {Object}
-     * @returns {Object}
-     *
-     **/
-    function cleanSettings(settingsMap) {
-        var ret = {};
-        for (var key in settingsMap) {
-            for (var k2 in settingsMap[key].userSettings) {
-                ret[k2] = settingsMap[key].userSettings[k2].name;
-                ret.channelLabel = key;
+    /**
+     * @param response {Object}
+     */
+    function authHandler(response) {
+        var promise = new Promise();
+        promise.then(
+            function(authMethod) {
+                log('log', "Request auth method successfull, the response containing " + authMethod + " is being sent", true);
+
+                response.status(200).json({
+                    v: 1,
+                    p: authMethod
+                });
+            },
+            function(reason, statusCode) {
+                statusCode = statusCode ? statusCode : 400;
+                log('log', "Request auth method unsuccessfull, the response containing the error message is being sent.", true);
+
+                response.status(statusCode).json(reason);
+            }
+        );
+
+        switch (streamType) {
+            case "public":
+                self.oauthClient.getAuthorizationUrl(function(err, url) {
```

Comment:
The REQ_AUTH event will be used only for private streams, therefore this 'switch' is unnecessary.